ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/AtomicActionAsyncTest.java

```diff
@@ -49,6 +55,42 @@ public void testAbort() throws Exception {
     }
 
     @Test
+    public void testAbortAsyncAC() throws Exception {
+        boolean[] testAbortAsyncAC = new boolean[]{true};
+        int[] called = new int[1];
+        int[] rollback = new int[1];
+
+        AtomicActionTestBase.init(false);
+        arjPropertyManager.getCoordinatorEnvironmentBean().setAsyncAfterSynchronization(true);
+
+        final Thread thread = Thread.currentThread();
+        executeTest(false, ActionStatus.ABORTED, new SyncRecord[]{new SyncRecord() {
+            @Override
+            public boolean afterCompletion(int status) {
+                if (status == ActionStatus.ABORTED) {
+                    rollback[0]++;
+                }
+                testAbortAsyncAC[0] = testAbortAsyncAC[0] && (Thread.currentThread() != thread);
```

Comment:
It looks like this test only validates that after completion callbacks run in new threads. If that is the case, I have three questions:

1. why not just say assertNotEquals(Thread.currentThread(),  thread) here and get rid of the testAbortAsyncAC and rollback variables, it makes the test clear.
2. why do you need to test for rollback, we have other tests that to that or is that the bug you have created the test for? But if you do need the check you could do it here: if (status != ABORTED) then fail("should have aborted")
3. it not clear how the test validates that the bug is fixed - please can add a comment to explain what the test is testing for